docs/framework/winforms/controls/how-to-open-files-using-the-openfiledialog-component.md

```diff
@@ -1,172 +1,237 @@
+
+To get or set the <xref:System.Windows.Forms.FileDialog.FileName%2A> property requires a privilege level granted by the <xref:System.Security.Permissions.FileIOPermission?displayProperty=nameWithType> class. The examples run a <xref:System.Security.Permissions.FileIOPermission> permission check, and can throw an exception due to insufficient privileges if run in a partial-trust context. For more information, see [Code access security basics](../../../../docs/framework/misc/code-access-security-basics.md).
+
+You can build and run these examples standalone from the C# or Visual Basic command line. For more information, see [Command-line building with csc.exe](../../../csharp/language-reference/compiler-options/command-line-building-with-csc-exe.md) or [Build from the command line](../../../visual-basic/reference/command-line-compiler/building-from-the-command-line.md). 
+
+For .NET Core 3.0, you can build and run the examples from a folder that has a Windows Forms *\<folder name>.csproj* file with the following contents: 
+```csharp
+<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
+  <PropertyGroup>
+    <OutputType>WinExe</OutputType>
+    <TargetFramework>netcoreapp3.0</TargetFramework>
+    <RootNamespace>folder name</RootNamespace>
+    <UseWindowsForms>true</UseWindowsForms>
+  </PropertyGroup>
+</Project>
+```
+
+## Example: Read a file as a stream with StreamReader  
+  
+The following example uses a <xref:System.Windows.Forms.Button> control's <xref:System.Windows.Forms.Control.Click> event handler to open the <xref:System.Windows.Forms.OpenFileDialog> with the <xref:System.Windows.Forms.CommonDialog.ShowDialog%2A> method. After the user chooses a file and selects **OK**, an instance of the <xref:System.IO.StreamReader> class reads the file and displays its contents in a message box. For more information about reading from file streams, see <xref:System.IO.FileStream.BeginRead%2A?displayProperty=nameWithType> and <xref:System.IO.FileStream.Read%2A?displayProperty=nameWithType>.  
+ 
+
+```csharp  
+using System;
+using System.IO;
+using System.Security;
+using System.Windows.Forms;
+
+public class OpenFileDialogForm : Form
+{
+    public Button selectButton;
+    public OpenFileDialog openFileDialog1;
+
+    public OpenFileDialogForm()
+    {
+        selectButton = new Button();
+        openFileDialog1 = new OpenFileDialog();
+
+        selectButton.Text = "Select file";
+        Controls.Add(selectButton);
+        selectButton.Click += new EventHandler(selectButton_Click);
+    }
+
+    [STAThread]
+    public static void Main()
+    {
+        Application.Run(new OpenFileDialogForm());
```

Comment:
```suggestion
        Application.EnableVisualStyles();
        Application.SetCompatibleTextRenderingDefault(false);
        Application.Run(new OpenFileDialogForm());
```
The two lines should also be added to Main in the second example for both VB and C#.
Also need to add this code to the second set of files.

Reply:
@rpetrusha - Application.SetCompatibleTextRenderingDefault(false); throws a runtime error unless it is the first thing in the file so I attempted to rearrange things.